arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
Sorry to chime in late here, but it sounds like we're going for a seemingly more secure, and definitely more surprising configuration.

Preventing writes to flash isn't really all that much of a security improvement. In order to actually have secure boot, the bootloader has to be protected by hardware. MPU is insufficient, because it is easy to break before it is set up and then just replace the main code.

Once the bootloader is running, there is no security issue with being able to write to the code space. The images must be signed in order to be bootable.

In addition, apps using the bootloader are going to want to do OTA, which will require flash writing.

I really don't see any scenario where the flash driver would be wanted, but writes not allowed to the flash device.